chore(layers): add release pipeline in GitHub Actions (#1278)
awslabs/aws-lambda-powertools-python: layer/README.md

<!-- markdownlint-disable MD041 MD043-->
# CDK Powertools layer

> This is a CDK project to build and deploy AWS Lambda Powertools layer to multiple commercial regions.

Reviewer: ```suggestion
This is a CDK project to build and deploy AWS Lambda Powertools Lambda Layer to multiple commercial regions.
```

## Build the layer

> To build the layer construct you need to provide the powertools version.

Reviewer: ```suggestion
To build the layer construct you need to provide the Powertools version that is [available in PyPi](https://pypi.org/project/aws-lambda-powertools/).
```

You can pass it as a context variable when running `synth` or `deploy`,

```shell
cdk synth --context version=1.25.1
```

## Canary stack

> We use a canary stack to verify that the deployment is successful and we can use the layer by adding it to a newly created Lambda function.

Reviewer: ```suggestion
We use a canary stack to verify that the deployment is successful and we can use the Layer by adding it to a newly created Lambda function.
```

Capital `L` for Layer.

Author: Why capital L 🤔? It's in lower case in the docs.

Reviewer: we should be fixing those then (rewriting examples now, then branding and doc rewrite). The reason is that `Layer` with capital L means a reference to a product or something (copywriting), while `layer` being a noun.

Author: No I mean `layer` is used in the [official docs](https://docs.aws.amazon.com/lambda/latest/dg/configuration-layers.html)

Reviewer: Ah! that's because they introduce it with `Lambda layer` and points to a glossary which we don't have. From there they continuously refer to Layer as it belongs to Lambda. Maybe we link to it and keep as `layer` then?

https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-concepts.html#gettingstarted-concepts-layer

> The canary is deployed after the layer construct. Because the layer ARN is created during the deploy we need to pass this information async via SSM parameter.

Reviewer: ```suggestion
The canary is deployed after the Layer construct. Because the Layer ARN is created during the deploy we need to pass this information async via SSM parameter.
```

> To achieve that, we establish a naming convention for the SSM key. The layer construct knows where to write the layer ARN after the deployment and the Canary stacks know where to read this information.

Reviewer: ```suggestion
To achieve that, we establish a naming convention for the SSM key. The Layer construct knows where to write the Layer ARN after the deployment and the Canary stacks know where to read this information.
```

> To achieve that, we establish a naming convention for the SSM key. The layer construct knows where to write the layer ARN after the deployment and the Canary stacks know where to read this information.

Reviewer: what is the naming convention?

Author: The name itself is actually irrelevant we just write and read ssm key. The word "convention" is probably misleading. I will write it.

## Version tracking

> AWS Lambda versions the layers by incrementing a number at the end of the arn.

Reviewer: ```suggestion
AWS Lambda versions Lambda Layers by incrementing a number at the end of the ARN.
```

> This means it's hard to tell which powertools version is inside the layer.

Reviewer: ```suggestion
This means it's hard to tell which Powertools version is inside the Layer.
```

Nit: `This makes it challenging to know which Powertools version a Layer contains.`

> For better tracking of the ARNs and the corresponding version we need to keep track which powertools version was deployed to which layer.

Reviewer: ```suggestion
For better tracking of the ARNs and the corresponding version we need to keep track which Powertools version was deployed to which Layer.
```

> To achieve that we created two components. First, we created a version tracking app which receives evnets via Event Bridge. Second, after a successful canary deployment we send the layer ARN, powertools version, and the region to this Event Bridge.

Reviewer: ```suggestion
To achieve that we created two components. First, we created a version tracking app which receives events via EventBridge. Second, after a successful canary deployment we send the layer ARN, Powertools version, and the region to this EventBridge.
```

EventBridge is the branded name
